### DISCUSSION TITLE
What's the advantage of using c interface

### DISCUSSION THREAD

**Discussion by waterdropw** [microsoft/onnxruntime#9311] [General]
onnxruntime interface is defined with c(onnxruntime_c_api.h), then wrapped with c++(onnxruntime_cxx_api.h) and the whole implementation is c++.
why not c++ interface? just for binary compatibility?  
Someone may concerned embeded device without C++ toolchain are still not supported.

**Comment by snnn** [comment]
1. Binary compatibility is a major concern. 
2. To support .Net P/Invoke

**Comment by pranavsharma** [comment]
C is the lowest common denominator that allows us to write bindings for several other languages (C#, Java, Rust, etc). Besides, binary compatibility is a strong requirement for us since many of our internal customers consume the ORT shared lib directly and (obviously) don't want to bother updating their source when they update ORT.

**Reply by waterdropw** [reply]
I see, thanks ^_^
